docs/contributing/contributing-rules.md

```diff
@@ -129,23 +84,286 @@ var numdata = 1;
+Use the YAML multiline string operator `>-` when rule messages span multiple lines. This presents the best-looking rule message on the command-line without having to worry about line wrapping or escaping the quote or backslash characters.
 
-<pre class="language-bash"><code>pipenv run ./cli/scripts/compare.py <span className="placeholder">VERSION_NUMBER OTHER_VERSION_NUMBER RULE_ID</span></code></pre>
+For an example of a good rule message, see: [this rule for Django's mark_safe](https://semgrep.dev/r?q=python.django.security.audit.avoid-mark-safe.avoid-mark-safe).
 
-Substitute the optional placeholder <code><span className="placeholder">VERSION_NUMBER OTHER_VERSION_NUMBER RULE_ID</span></code> with the respective version numbers and a rule ID you want to compare. 
+:::info Rule message example
+`mark_safe()` is used to mark a string as *safe* for HTML output. This disables escaping and may expose the content to XSS attacks. Instead, use `django.utils.html.format_html()` to build HTML for rendering.
+:::
+
+### Including additional details with rule metadata
+
+Rules in the Semgrep Registry require specific metadata fields that ensure consistency across the ecosystem in both Semgrep App and Semgrep CLI. These fields help users of Semgrep to identify rules in different categories such as:
+
+- High confidence security rules for CI pipelines.
+- OWASP Top 10 or CWE Top 25 rule packs.
+- Technology. For example `react` so it is easy to find `p/react` rule packs.
+- Audit rules with lower confidence intended for code auditors.
+
+Nest these metadata under the `metadata` key. The following metadata are required:
+
+<table>
+  <thead><tr>
+   <th>Metadata field</th>
+   <th>Possible values</th>
+   <th>Example use</th>
+  </tr></thead>
+  <tbody>
+  <tr>
+   <td>CWE</td>
+   <td>A <a href="https://cwe.mitre.org/index.html">Comment Weakness Enumeration (CWE)</a>.</td>
+   <td><code>cwe: "CWE-502: Deserialization of Untrusted Data"</code></td>
+  </tr>
+  <tr>
+   <td>Confidence</td>
+   <td><code>HIGH</code>, <code>MEDIUM</code>, <code>LOW</code></td>
+   <td><code>confidence: MEDIUM</code></td>
+  </tr>
+  <tr>
+   <td>Likelihood</td>
+   <td><code>HIGH</code>, <code>MEDIUM</code>, <code>LOW</code></td>
+   <td><code>likelihood: MEDIUM</code></td>
+  </tr>
+  <tr>
+   <td>Impact</td>
+   <td><code>HIGH</code>, <code>MEDIUM</code>, <code>LOW</code></td>
+   <td><code>impact: HIGH</code></td>
+  </tr>
+  <tr>
+   <td>Subcategory</td>
+   <td><code>vuln</code>, <code>audit</code>, <code>guardrail</code></td>
+   <td><code>subcategory:
+  - vuln</code></td>
+  </tr>
+  <tr>
+   <td>References</td>
+   <td>Any additional information that gives more context to the user of the rule. This helps to specify rule packs in Semgrep Registry.</td>
+   <td><code>references:
+      - react</code></td>
+  </tr>
+  </tbody>
+</table>
```

Comment:
@LewisArdern how do you like this table? Is it fine in its current state?

<img width="896" alt="Screenshot 2022-11-09 at 18 00 00" src="https://user-images.githubusercontent.com/26669307/200893402-3d806064-c585-4fc4-a65c-a40fa858acfe.png">